stl/inc/xmemory

```diff
@@ -2140,10 +2140,10 @@ template <class _Ty, class _Alloc, class _Uty1, class _Uty2, enable_if_t<_Is_spe
 _NODISCARD constexpr auto uses_allocator_construction_args(const _Alloc& _Al, pair<_Uty1, _Uty2>&& _Pair) noexcept {
     // equivalent to
     // return _STD uses_allocator_construction_args<_Ty>(_Al, piecewise_construct,
-    //     _STD forward_as_tuple(_STD move(_Pair).first), _STD forward_as_tuple(_STD move(_Pair).second));
+    //     _STD forward_as_tuple(get<0>(_STD move(_Pair)), _STD forward_as_tuple(get<1>(_STD move(_Pair)));
     return _STD make_tuple(piecewise_construct,
-        _STD uses_allocator_construction_args<typename _Ty::first_type>(_Al, _STD move(_Pair).first),
-        _STD uses_allocator_construction_args<typename _Ty::second_type>(_Al, _STD move(_Pair).second));
+        _STD uses_allocator_construction_args<typename _Ty::first_type>(_Al, get<0>(_STD move(_Pair))),
+        _STD uses_allocator_construction_args<typename _Ty::second_type>(_Al, get<1>(_STD move(_Pair))));

```

Comment:
Does the change effectively make the following:
```
forward_as_tuple(move(move(_Pair).fist)), forward_as_tuple(move(move(_Pair).second)));
```
?

If so, it could be better implemented with actual `move` if #1208 is somehow addressed.

Reply:
Yes, I think it is `move(move(_Pair).fist)` but I'm unsure if  it would be better. 

You may be right, I just don't know enough about it.

I was just looking at the issues that Stephan forgot to mark as "good first issue" 🙂

Reply:
Sure it is safer to have "good first issue"-ish implementation and do literally what LWG resolution asks :-)

Reply:
I don't believe that they would be equivalent. `pair` can contain lvalue references, so given `pair<int&, int&>`, `get<0>(move(p))` is `int&`, but `move(move(p).first)` would be `int&&`.